proto/substrait/relations.proto

```diff
@@ -86,6 +86,15 @@ message ReadRel {
 
       FileFormat format = 5;
 
+      // the index of the partition this item belongs to
+      uint64 partition_index = 6;
+
+      // the start position in byte to read from this item
+      uint64 start = 7;
+      
+      // the length in byte to read from this item
+      uint64 length = 8;
```

Comment:
Could we set these fields to be optional to avoid this issue?

Reply:
Proto fields are already optional. `proto3` support for explicit optional is pretty nascent and it's not likely that many systems are prepared to handle `optional` in `proto3` yet. There are some very important details around the meaning of field presence here https://github.com/protocolbuffers/protobuf/blob/v3.12.0/docs/field_presence.md that we'd need to be comfortable with before introducing this IMO.

Reply:
If the user doesn't specify slicing parameters then we assume they are reading the whole file.  The defaults match up with this nicely.

`start` - A default of 0 is what you would put in if you were explicitly reading the whole file anyways.
`length` - This is a more interesting case but we are in luck because 0 is an invalid length so we just need to document that 0 means "read the whole file"
`partition_index` - If they are going to read the whole file then calling it partition index 0 of 0 seems fine to me.  Nothing needed here.

Reply:
Sounds great.

Reply:
Thanks, LGTM!

Reply:
By the way, I did add default behavior explanation for `length` to the proto in https://github.com/substrait-io/substrait/pull/109/files#diff-cc42d548188fb7975a0eae621be239983723002ff219559a7876112d4fa78894R101 (though it seems that PR needs a rebase)